### PR TITLE
Added the cro reports to es

### PR DESCRIPTION
**Pull request by athiruma**
## Type of change
Note: Fill **x** in []
- [ ] bug
- [ ] enhancement
- [x] documentation
- [x] dependencies

## Description

1. Added the Azure reports to es.
2. Added the single cro_run file making it generic by calling the cro_object class, which has methods that return CRO classes.

### CODE REVIEW

File: cloud_governance/cloud_resource_orchestration/clouds/azure/resource_groups/collect_cro_reports.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+import typeguard
+
+from cloud_governance.cloud_resource_orchestration.clouds.azure.resource_groups.cost_over_usage import CostOverUsage
+from cloud_governance.cloud_resource_orchestration.clouds.common.abstract_collect_cro_reports import \
+    AbstractCollectCROReports
+from cloud_governance.common.logger.logger_time_stamp import logger_time_stamp
+from cloud_governance.main.environment_variables import environment_variables
+
+
+class CollectCROReports(AbstractCollectCROReports):
```
Comment (ebattat) [inline cloud_governance/cloud_resource_orchestration/clouds/azure/resource_groups/collect_cro_reports.py:13]: Same question, why we need separate CRO class for Azure ?

File: cloud_governance/cloud_resource_orchestration/common/cro_object.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
+import boto3
+
+from cloud_governance.cloud_resource_orchestration.clouds.aws.ec2.aws_monitor_tickets import AWSMonitorTickets
+from cloud_governance.cloud_resource_orchestration.utils.common_operations import string_equal_ignore_case
+from cloud_governance.main.environment_variables import environment_variables
+
+
+class CroObject:
```
Comment (ebattat) [inline cloud_governance/cloud_resource_orchestration/common/cro_object.py:8]: Missing documentation

File: cloud_governance/cloud_resource_orchestration/common/run_cro.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+from cloud_governance.cloud_resource_orchestration.common.cro_object import CroObject
+from cloud_governance.cloud_resource_orchestration.utils.common_operations import string_equal_ignore_case
+from cloud_governance.common.elasticsearch.elasticsearch_operations import ElasticSearchOperations
+from cloud_governance.common.logger.init_logger import logger
+from cloud_governance.common.logger.logger_time_stamp import logger_time_stamp
+from cloud_governance.main.environment_variables import environment_variables
+
+
+class RunCRO:
+    """This class monitor cro activities"""
```
Comment (ebattat) [inline cloud_governance/cloud_resource_orchestration/common/run_cro.py:12]: monitors

File: cloud_governance/cloud_resource_orchestration/common/run_cro.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+                f'last_run_{self.__account}': datetime.utcnow()}, id=self.PERSISTENT_RUN_DOC_ID)
+        else:
+            self.__es_operations.update_elasticsearch_index(index=self.PERSISTENT_RUN_INDEX,
+                                                            metadata={f'last_run_{self.__account}': datetime.utcnow()},
+                                                            id=self.PERSISTENT_RUN_DOC_ID)
+
+    @logger_time_stamp
+    def __send_cro_alerts(self):
+        """
+        This method send the cost_over_usage alert and Ticket status alerts
```
Comment (ebattat) [inline cloud_governance/cloud_resource_orchestration/common/run_cro.py:46]: sends

File: cloud_governance/cloud_resource_orchestration/common/run_cro.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+            logger.info(f"""{active_region}: -> Running CRO Resource data Collection""")
+            monitor_response = cro_monitor.run()
+            if monitor_response:
+                cro_reports = self.__cro_reports.run(monitor_response)
+                logger.info(f'Cloud Orchestration Resources: {cro_reports}')
+
+    @logger_time_stamp
+    def __start_cro(self):
+        """
+        This method start the cro process methods
```
Comment (ebattat) [inline cloud_governance/cloud_resource_orchestration/common/run_cro.py:95]: starts

**Review by ebattat — COMMENTED** [review]
I think that we can create same abstract interface class for AWS and Azure

File: cloud_governance/cloud_resource_orchestration/clouds/azure/resource_groups/collect_cro_reports.py (hunk 1 above)
Comment (athiruma) [reply cloud_governance/cloud_resource_orchestration/clouds/azure/resource_groups/collect_cro_reports.py:13]: yes, the interface is same

File: cloud_governance/cloud_resource_orchestration/clouds/azure/resource_groups/azure_monitor_tickets.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    IN_PROGRESS = 'In Progress'
+    CLOSE_JIRA_TICKET = 0
+    FIRST_CRO_ALERT: int = 5
+    SECOND_CRO_ALERT: int = 3
+    DEFAULT_ROUND_DIGITS: int = 3
+
+    def __init__(self, region_name: str = ''):
+        super().__init__()
+
+    # Todo All the below methods will be implement in future releases
```
[the remark this answers is not in the document]
Comment (athiruma) [reply cloud_governance/cloud_resource_orchestration/clouds/azure/resource_groups/azure_monitor_tickets.py:40]: This is the Todo statement

File: cloud_governance/cloud_resource_orchestration/clouds/azure/resource_groups/azure_monitor_tickets.py [hunk 7]
```diff
@@ -0,0 +1,66 @@
+
```
[the remark this answers is not in the document]
Comment (athiruma) [reply cloud_governance/cloud_resource_orchestration/clouds/azure/resource_groups/azure_monitor_tickets.py:1]: yes, same interface class used

**Review by ebattat — COMMENTED** [review]
You have a lot of common folders, need to verify it that correct approach

**Comment by athiruma** [comment]
> You have a lot of common folders, need to verify it that correct approach

Yeah sure.
May be I need to re-arrange the files.

**Comment by ebattat** [comment]
@athiruma, are we ok with the nested common folder ?

**Comment by athiruma** [comment]
> @athiruma, are we ok with the nested common folder ?

I am ok with this. Will be renamed it according to the requirements.

**Review by ebattat — APPROVED** [review]
/LGTM

**Pull request merged**